server/image_pull.go

```diff
@@ -69,6 +69,7 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 		if !inProgress {
 			pullOp = &pullOperation{}
 			s.pullOperationsInProgress[pullArgs] = pullOp
+			storage.ImageBeingPulled.Store(pullArgs.image, 1)
```

Comment:
I want to highlight one _potential_ error source. `pullArgs` include the credentials, so it is possible that the same image is being pulled by more than one routine. That's why I suggested an `int` which would allow for reference counting.

Looking at the PR now, I think that reference counting isn't necessary, so we could change `int` to `bool` and squeeze a bit of memory.